components/encoder/fake/encoder.go

```diff
@@ -82,11 +85,11 @@ type fakeEncoder struct {
 	resource.Named
 	resource.TriviallyCloseable
```

Comment:
You've implemented `Close`. This is no longer `TriviallyCloseable` (and given that it's had a background goroutine this whole time, I don't think it ever should have been! :eyes:)